libraries/AP_Airspeed/AP_Airspeed_ASP5033.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+  backend driver for airspeed from www.qio-tek.com a I2C ASP5033 sensor
+ */
+
+#include "AP_Airspeed_ASP5033.h"
+#include <GCS_MAVLink/GCS.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_HAL/I2CDevice.h>
+#include <AP_Math/AP_Math.h>
+
+extern const AP_HAL::HAL &hal;
+
+#define ASP5033_I2C_ADDR 0x6D
+#define Sys_config 0xA5
+#define CMD 0x30
+#define CMD_measure 0x0A
+#define Press_data 0x06
+#define Temp_data 0x09
+
+
+AP_Airspeed_ASP5033::AP_Airspeed_ASP5033(AP_Airspeed &_frontend, uint8_t _instance) :
+    AP_Airspeed_Backend(_frontend, _instance)
+{
+}
+
+// probe and initialise the sensor
+bool AP_Airspeed_ASP5033::init()
+{
+    uint8_t SysID;
+    _dev = hal.i2c_mgr->get_device(get_bus(), ASP5033_I2C_ADDR);
+    if (!_dev) {
+        return false;
+    GCS_SEND_TEXT(MAV_SEVERITY_INFO,"ASP5033: sensor no found\n");
+    }
+    _dev->get_semaphore()->take_blocking();
+    _dev->set_retries(2);
+    _dev->get_semaphore()->give();
+    _dev->read_registers(Sys_config, &SysID, 1); //Read Sensor_config
```

Comment:
Is there any type of WHOAMI we can read from the sensor? I'm mostly just wondering if we have a way to derisk accidentally detecting a FBM320 baro as the airspeed sensor.

Reply:
This is a custom sensor.

Reply:
@QioTek @WickedShell is concerned that we may not be able to distinguish this from the FBM320

Can you add a `WHOAMI`?

Reply:
Added